autolab/Autolab
# Fix scoreboard sorting failing on custom scoreboardOrderSubmissions, add more error checking when creating scoreboard entries / displaying

## Description


- Adds more error-catching logic when calling custom `scoreboardOrderSubmissions`, in the case that an exception is raised, and tells the instructor via an error message
- Adds a rescue block to the top-level sort call in `#show` so that if a bad sort / comparison occurs, the error can be caught
- Instead of returning scoreboard entries with "-" on errors, raise an error, and in `#show`, revert to an empty array
- In the `#show` view code, check that the current entry length is greater than the index in order to show a score, else show an error icon (which handles the empty array case)
- Fix up logging, add more specific errors to be returned to instructors, and redirect students back to assessments index

## Motivation and Context



Closes #2086 

## How Has This Been Tested?

Import the provided assessment:
[randomlab5_20240217.tar.zip](https://github.com/autolab/Autolab/files/14320229/randomlab5_20240217.tar.zip)

1. Upload at least two submissions as students
2. View scoreboard, see that it renders fine (if it is not rendering fine, upload `autograder.tar`as the autograder)
3. Upload the assessment config file `randomlab5-bad-scoreboard-order.rb` (in Edit Assessment you can upload)
  - this test returns a bad sort, causing an ArgumentError
5. See that you get an error as follows:
![Screenshot 2024-02-17 at 3 22 59 PM](https://github.com/autolab/Autolab/assets/25730111/7b66661a-439f-464d-929c-d47805d7e426)
6. Try to view scoreboard as a student and see that you get redirected back to assessment index with an error flash
7. Upload the initial assessment config file back, and then upload `autograde-bad.tar` as the new autograder
 - this autograder returns just a number instead of an array
9. Regrade all submissions
10. view scoreboard as instructor, see that following error appears:
![Screenshot 2024-02-17 at 3 59 06 PM](https://github.com/autolab/Autolab/assets/25730111/a23887cf-ef99-4f2e-80ef-0e1d3b28938d)
11. View scoreboard as student, see that scoreboard is visible but error icons appear:
![Screenshot 2024-02-17 at 4 03 59 PM](https://github.com/autolab/Autolab/assets/25730111/18ed6c9e-4259-4cea-9e98-c9421b9f2265)
12. Upload the old autograder.tar back, and regrade all
13. Then add this code after line 296 in `scoreboards_controller.rb`:
```
if [true, false].sample
      parsed["scoreboard"] = 0
    end
```
( you may need to also reload the config file)
  - this replicates the scenario where sometimes the scoreboard results aren't correct / autograder failed
14. view the scoreboard as an instructor, see the following error
![Screenshot 2024-02-17 at 4 41 53 PM](https://github.com/autolab/Autolab/assets/25730111/59c4c601-641e-42d7-98e7-7c3bc37c8cab)

15. As a student, see something like this (all the bad scoreboard entries are at the bottom)
![Screenshot 2024-02-17 at 2 30 28 PM](https://github.com/autolab/Autolab/assets/25730111/3bf4a73c-1d8b-44dc-8fff-56d600d049b4)

## Types of changes
